BlazarUI/app/scripts/components/shared/RepoBranchCardStack.jsx

```diff
@@ -0,0 +1,67 @@
+import React, {Component, PropTypes} from 'react';
+import Immutable from 'immutable';
+import {CardStack} from 'card-stack-test';
+
+
+import RepoBranchCard from './RepoBranchCard.jsx';
+import RepoBranchCardStackHeader from './RepoBranchCardStackHeader.jsx';
+import RepoBranchCardStackZeroState from './RepoBranchCardStackZeroState.jsx';
+
+class RepoBranchCardStack extends Component {
+
+  shouldComponentUpdate(nextProps) {
+    return JSON.stringify(this.props.starredBuilds.toJS()) !== JSON.stringify(nextProps.starredBuilds.toJS())
+      || this.props.expandedCard !== nextProps.expandedCard
+      || JSON.stringify(this.props.moduleBuildsList) !== JSON.stringify(nextProps.moduleBuildsList);
+  }
+
+  renderCards() {
+    const numberOfBuilds = this.props.starredBuilds.size;
+
+    return this.props.starredBuilds.map((build, key) => {
+      return (
+        <RepoBranchCard
```

Comment:
onClick, expanded, belowExpanded and loading should be added in by the card stack